packages/ember-runtime/lib/system/array_proxy.js

```diff
@@ -373,6 +373,17 @@ var ArrayProxy = EmberObject.extend(MutableArray, {
     this._super(...arguments);
     this._setupContent();
     this._setupArrangedContent();
+
+    if (!this._LEGACY_EMBER_DATA_SUPPORT) {
```

Comment:
I believe this is correct.  Basically, we are only deprecating when not used via Ember Data.  This is on a per instance basis, not a global ENV basis.